cluster/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingContextFactory.scala

```diff
@@ -61,13 +56,18 @@ class SnappyStreamingContextFactory extends SparkContextFactory {
     new SnappyStreamingContext(LeadImpl.getInitializingSparkContext,
       Milliseconds(interval)) with ContextLike {
 
+      private val addedJars = scala.collection.mutable.ArrayBuffer.empty[String]
```

Comment:
@rishitesh Does this hold for the new Spark streaming too? There is no single streaming context, rather new source/sink registered as required. Perhaps it requires a new job interface, but if addJobJar is used there too then it will cause trouble (which will certainly be hard to detect for someone just using the method). Considering that not-so-future case this should be made thread-safe assuming it can be trouble then.
@ymahajan